### DISCUSSION TITLE
Create toggle to disable "Convert to issue" from Markdown checkboxes feature

### DISCUSSION THREAD

**Discussion by HonkingGoose** [community/community#4261] [Projects and Issues]
GitHub now offers to "convert to issue" on Markdown checkbox items when you hover over a checkbox item.
I don't find this very useful, as you can mis-click and end up with a new issue.
Also the hover state is distracting, when you're just mousing over things.

Can we pretty please get a toggle to turn this feature off completely, preferably on a a username/organization level.

I help triage the Renovate bot repository, and our issue/pull request views now lists "tasks" for each issue/PR (displayed) as partially complete, I find this really distracting, as those issues/PRs are not using "todo"-checkboxes, but use checkboxes as a way to make data-entry on required fields easier.

I'd like a way to opt-out/toggle this behavior on a repo/user/org based level.

**Comment by 0xdevalias** [comment]
Further to this, it might actually be really useful if we could have a way to disable it in a 'local' way from within the markdown code itself. Eg. I love this feature generally, but we have some 'did you complete all the tasks when creating this issue' checklists in our issue templates that I would love to be able to disable it for.

**Reply by HonkingGoose** [reply]
In my experience people are _really_ happy to throw out the provided issue template / pull request template and do their own thing. 😉 

Suppose you had a comment in the Markdown like this:

```markdown
<-- Comment that disables convert to issue checkboxes -->

Start of template
```

People would still remove those comments, and do their own thing.

```markdown
Start of template
```

I like the simplicity of your proposal, but real world users _will_ probably mess things up. 😄 

---

> we have some 'did you complete all the tasks when creating this issue' checklists

I _highly_ recommend you take a look at the new GitHub Issue _Forms_. This is basically a way to provide users with a standardized form. People cannot mess with the form, and you can require certain entries to be filled out. This should reduce instances of people blatantly ignoring your templates... 😉 

This is a beta feature, so it's a bit risky, but I find the new Forms easy to use. It's a bit tricky to get started with the YAML syntax though.

Documentation here:

- https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms
- https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema

**Comment by azenMatt** [comment]
Do you find the general concept useful? 

How would you feel if this was re-designed to be much harder to trigger accidentally?

**Reply by HonkingGoose** [reply]
> Do you find the general concept useful?

Not really, I usually create issues by hand for everything that needs to be done. But I think the general idea is still a good one, but the end-user should have more control over if the feature is active or not on a issue/repo/organization level.

> How would you feel if this was re-designed to be much harder to trigger accidentally?

That would be a good first step! It's way too easy to trigger this behavior on accident right now.

---

To clarify why I would want to turn this off completely for a issue, consider the Renovate bot Dependency Dashboard workflow.
Renovate bot helps users to update their dependencies (like Dependabot). Renovate bot users can enable a "Dependency Dashboard Issue", that lists all current/pending/scheduled updates for easy overview.

Within the Renovate Dependency Dashboard issue, you can click on a markdown checkbox. This tells Renovate bot: "give me a PR with this update right now".

The problem with the "convert to issue" feature is that you can create new issues from the dashboard by accident. Those issues make zero sense to anybody.

I'd like a way to tell GitHub: Do not use the "Convert to issue" on markdown checkboxes for this issue (maybe via a Markdown comment).

Here's an example of the Renovate Dependency Dashboard issue: https://github.com/renovatebot/renovate/issues/2958

**Reply by ADTC** [reply]
Concept could be useful, but the assumption that _all lists are task lists_ is **100% flawed** and shows a lack of understanding of how the community uses lists. Please provide a way to control this.

**Reply by richardeakin** [reply]
FWIW I find the concept very useful, and is something I found missing and exists in other issue trackers like Jira. Just a matter of honing in on the UX.

**Comment by avioli** [comment]
First - an observation:

A "list" with no space between brackets:
[] a
[] b

---

A proper list with no space between brackets:
 - [] a
 - [] b

NOTE: While editing this list continues to suggest new list entries after creating a new line at the end of a task list entry.

---

A list with space between brackets - making it a task list with no pre-checked tasks:
 - [ ] a
 - [ ] b

NOTE: While editing this list continues to suggest new **task** list entries after creating a new line at the end of a task list entry.

---

A list with `x` between brackets - making it a task list with pre-checked tasks:
 - [x] a
 - [x] b

NOTE: While editing this list continues to suggest new **non-checked task** list entries after creating a new line at the end of a task list entry.

---

A list with `X` between brackets - making it a task list with pre-checked tasks:
 - [X] a
 - [X] b

NOTE: While editing this list continues to suggest new list entries after creating a new line at the end of a task list entry.

---

All the above lists are defined, using a `-` for a list entry prefix, but use of `*` is also an option.

Here is a screenshot of the above representations, once rendered by GitHub:

<img width="86" alt="Screen Shot 2021-08-09 at 10 26 49 am" src="https://user-images.githubusercontent.com/524259/128650339-f0ef5f08-1e8d-4b1a-8b1e-eaad665a9cac.png">

---

@azenMatt 

Having observed the above options I would suggest to **only** add the option to "Convert to issue" when using "empty" checkboxes - `[ ]`, the lowercase `[x]` or a mix of both, which will cover the most use-cases at the moment.

For the case where users can "opt-out" - GitHub could ignore use of the two alternatives - two brackets `[]` or the capital `[X]`. This suggestion will require GitHub to also convert `- [] text` and `- [] text` to un-checked list entries.

My last suggestion is that a mix use of `[ ]`, `[x]` with `[]`, `[X]` should be considered the latter case.

Given the editor allows issues and comments to be edited, a user can easily convert to the alternative version.

**Reply by HonkingGoose** [reply]
I don't think this distinction between lowercase `x` and uppercase `X` matters. Both will return a GitHub Markdown checkbox that people can click. I've seen people use lowercase or uppercase interchangeably. I don't think people will remember to use the correct type of casing.

It's not really about users opting in/out, it's about repository maintainers picking what behavior they want. I don't see _why_ the person who opens the issue should affect the Markdown checkboxes convert to issue behavior.

Maybe I'm missing something that explains why you want to go this route?

**Reply by avioli** [reply]
I was giving just a suggestion, even if a bit elaborate.

I've used both lower and uppercase myself, but tend to prefer the lowercase, which is what GitHub sets the value to when you check one such task list item. Based on that, I thought, the uppercase could be used for regular old-school checkbox lists that are not backed up by the new feature, which means the "Convert to issue" will not show up. That's all.

**Reply by azenMatt** [reply]
It feels like we have two issues here that are slightly overlapping. 

1. It is too easy to accidentally create a new issue from a task list, which increases noise 
2. There are many cases where checklists are used but the desired end state is _not_ to have all of them selected. In this case seeing `x of y` is both distracting and misleading 

One proposed solution would be to allow the feature to be turned off at the repo/org level, but I'd love to try and solve them in product first. 


A couple of thoughts: 

For problem 1. Have you found value in this feature in any use cases? I am wondering if it just needs some design tweaks to make it more deliberate? 

For problem 2. What do you think about allowing users to 'opt-out' on a per issue bases. Essentially generating the `x of y` in the UI, but allowing you to dismiss it in someway to say 'the goal here isn't to get to 100% completion'?

**Reply by avioli** [reply]
@azenMatt Even though task lists are one feature it contains three aspects (that I observed):

  1. Having dynamically persisted checkboxes that other users with write access to the repo can check and uncheck;
  2. An option for other users with write access to the repo to convert tasks to issues;
  3. Issues that have task lists outline the said task list's completion state - `x of y`.

My thoughts on the two problems you outlined are:

  1. I don't believe the creation of new issue from a task is accidental. Maybe once, but then people will know what that aspect does. I do see the value, especially if a task list is meant to outline the issues that need to be worked on. The problem I think @HonkingGoose has is that not every list is a task list that is meant to be used for creating issues, which I feel is in an indirect way what you define in problem 2.
  2. Adding an option to opt-out from supporting "tasks' completion state" - `x of y` as you define it would be great and I think it should be on a per-issue basis, but also on a per-repo level. I believe that will sort out any and all issues people may have with the third aspect I outlined above. A new question pops straight up - should this aspect be an opt-in instead? Then another - how do you opt-in or opt-out?

I almost feel like these features should be some modern custom-element that users can embed, but GitHub provides or even repo owners can (install and) enable/disable:

```text
<task-list convert-to-issues="true" track-status="true">
 - [ ] task one
 - [ ] task two
</task-list>
```

or even:

```text
[task-list convert-to-issues="true" track-status="true"]
 - [ ] task one
 - [ ] task two
[/task-list]
```

**Comment by chengtbf** [comment]
how to disable github issue checklist “convert to issue” button？

**Comment by Snapperfish** [comment]
Use Firefox. https://github.com/github-community/community/discussions/5643
